terraform/elasticsearch.tf.json.template.py

```diff
@@ -11,7 +11,15 @@
 
 domain = config.es_domain
 
-emit_tf(None if config.share_es_domain else {
+emit_tf({
+            "data": {
```

Comment:
Since you reintroduced the methods in `deployment.py`, we can use those instead of the data source, I think.

Reply:
True, we could go back to using boto for provisioning the ES-Endpoint for deployments sharing ES instance with `dev`. This would require adding conditional logic in `prepare_lambda_deployment` to determine if it should use Terraform syntax or the appropriate method in `deployment.py`. This approach will add additional web requests when packaging lambdas, when it could all be done when Terraform is updating/publishing resources. For those reasons I think we should delegate that task to Terraform instead, since it is the authoritative entity for managing the ES resource. However, if you don't think this reasons are convincing enough I can revert to using `deployment.py` instead, as you emphasize.